File: frontend/src/types/trading-calendar.ts
/** 交易日历 API 类型 */

export interface TradingCalendarStatus {
  source: string
  last_synced_at: string | null
  row_count: number
  min_date: string | null
  max_date: string | null
  last_error: string | null
  degraded: boolean
  today_is_trade_day: boolean
  data_trade_date: string
  missing_sources: string[]
}

export interface TradingCalendarResolve {
  input_date: string
  trade_date: string
}

File: backend/app/schemas/trading_calendar.py
"""交易日历 API DTO。"""

from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, Field


class TradingCalendarStatus(BaseModel):
    source: str = "akshare_sina"
    last_synced_at: datetime | None = None
    row_count: int = 0
    min_date: date | None = None
    max_date: date | None = None
    last_error: str | None = None
    degraded: bool = True
    today_is_trade_day: bool = False
    data_trade_date: date
    missing_sources: list[str] = Field(default_factory=list)


class TradingCalendarResolve(BaseModel):
    input_date: date
    trade_date: date
